src/Servers/IIS/IIS/test/testassets/InProcessWebSite/Startup.cs

```diff
@@ -1030,5 +1032,153 @@ public Task InvalidCharacter(HttpContext context)
+        {
+            Assert.Equal("HTTP/2", context.Request.Protocol);
+            Assert.True(context.Response.SupportsTrailers());
+            return Task.FromResult(0);
+        }
+
+        public Task ResponseTrailers_ProhibitedTrailers_Blocked(HttpContext context)
+        {
+            Assert.True(context.Response.SupportsTrailers());
+            foreach (var header in DisallowedTrailers)
+            {
+                Assert.Throws<InvalidOperationException>(() => context.Response.AppendTrailer(header, "value"));
+            }
+            return Task.FromResult(0);
+        }
+
+        public Task ResponseTrailers_NoBody_TrailersSent(HttpContext context)
+        {
+            context.Response.DeclareTrailer("trailername");
+            context.Response.AppendTrailer("trailername", "TrailerValue");
+            return Task.FromResult(0);
+        }
+
+        public async Task ResponseTrailers_WithBody_TrailersSent(HttpContext context)
+        {
+            await context.Response.WriteAsync("Hello World");
+            context.Response.AppendTrailer("TrailerName", "Trailer Value");
+        }
+
+        public async Task ResponseTrailers_WithContentLengthBody_TrailersSent(HttpContext context)
+        {
+            var body = "Hello World";
+            context.Response.ContentLength = body.Length;
+            await context.Response.WriteAsync(body);
+            context.Response.AppendTrailer("TrailerName", "Trailer Value");
+        }
+
+        public async Task ResponseTrailers_WithTrailersBeforeContentLengthBody_TrailersSent(HttpContext context)
+        {
+            var body = "Hello World";
+            context.Response.ContentLength = body.Length * 2;
+            await context.Response.WriteAsync(body);
+            context.Response.AppendTrailer("TrailerName", "Trailer Value");
+            await context.Response.WriteAsync(body);
+        }
+
+        public async Task ResponseTrailers_WithContentLengthBodyAndDeclared_TrailersSent(HttpContext context)
+        {
+            var body = "Hello World";
+            context.Response.ContentLength = body.Length;
+            context.Response.DeclareTrailer("TrailerName");
+            await context.Response.WriteAsync(body);
+            context.Response.AppendTrailer("TrailerName", "Trailer Value");
+        }
+
+        public async Task ResponseTrailers_WithContentLengthBodyAndDeclaredButMissingTrailers_Completes(HttpContext context)
+        {
+            var body = "Hello World";
+            context.Response.ContentLength = body.Length;
+            context.Response.DeclareTrailer("TrailerName");
+            await context.Response.WriteAsync(body);
+        }
+
+        private TaskCompletionSource<object> _trailersReceivedForBody = new TaskCompletionSource<object>();
+        public async Task ResponseTrailers_CompleteAsyncWithBody_TrailersSent(HttpContext context)
+        {
+            await context.Response.WriteAsync("Hello World");
```

Comment:
I think this function is never called? And/or, if it is, `_trailersReceivedForBody .Task` would never complete right?